test/jasmine/tests/legend_scroll_test.js

```diff
@@ -96,6 +97,69 @@ describe('The legend', function() {
                 'translate(0, ' + finalDataScroll + ')');
         });
 
+        function dragScroll(element, rightClick) {
+            var scrollBox = getScrollBox();
+            var scrollBar = getScrollBar();
+            var legendHeight = getLegendHeight(gd);
+            var scrollBoxYMax = gd._fullLayout.legend._height - legendHeight;
+            var scrollBarYMax = legendHeight -
+                constants.scrollBarHeight -
+                2 * constants.scrollBarMargin;
+            var initialDataScroll = scrollBox.getAttribute('data-scroll');
+            var dy = 50;
+            var finalDataScroll = '' + Lib.constrain(initialDataScroll -
+                dy / scrollBarYMax * scrollBoxYMax,
+                -scrollBoxYMax, 0);
+
+            var scrollBarBB = scrollBar.getBoundingClientRect();
+            var y0 = scrollBarBB.top + scrollBarBB.height / 2;
+            var y1 = y0 + dy;
+
+            var elBB = element.getBoundingClientRect();
+            var x = elBB.left + elBB.width / 2;
+
+            var opts = {element: element};
+            if(rightClick) {
+                opts.button = 2;
+                opts.buttons = 2;
+            }
+
+            mouseEvent('mousedown', x, y0, opts);
+            mouseEvent('mousemove', x, y1, opts);
+            mouseEvent('mouseup', x, y1, opts);
+
+            expect(finalDataScroll).not.toBe(initialDataScroll);
+
+            return finalDataScroll;
+        }
+
+        it('should scroll on dragging the scrollbar', function() {
```

Comment:
Are these general legend scrollbox tests and/or tests that :lock: the fix for #2387? (I'm asking as I would be surprised if #2387 was reproducible in a jasmine test).

Reply:
Yes, these do test #2387, if I'm interpreting it correctly. What I think was happening in #2387 is that *somehow* the [`scrollBox` drag](https://github.com/plotly/plotly.js/pull/2426/commits/f75f5ba325e7c80d93272b71585d0e31ceac4f48#diff-f48c57dae9095e916f13f364ae740f69L296) was being triggered - and that's what I decided (with no dissent) had no reason ever to happen. So this test shows that regular drag works, then the [next one](https://github.com/plotly/plotly.js/pull/2426/commits/f75f5ba325e7c80d93272b71585d0e31ceac4f48#diff-ed9e861dab1a2e0cbdeacb9c7bd74db1R145) shows that the `scrollBox` won't drag even if it does somehow get the event, and then the [third new test](https://github.com/plotly/plotly.js/pull/2426/commits/f75f5ba325e7c80d93272b71585d0e31ceac4f48#diff-ed9e861dab1a2e0cbdeacb9c7bd74db1R154) shows that right-click (the only form of #2387 I could reproduce on my own computer) even on the `scrollBar` will not trigger drag.